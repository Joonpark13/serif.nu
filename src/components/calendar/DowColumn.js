import React from 'react';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/styles';
import Typography from '@material-ui/core/Typography';
import {
  HOURS,
  columnBorderStyle,
  columnMinHeight,
  columnHeight,
} from './calendar-constants';
import HourCell from './HourCell';

const useStyles = makeStyles({
  dowColumn: {
    flexGrow: 1,
    borderRight: columnBorderStyle,
    minHeight: columnMinHeight,
  },
  dowHeader: {
    borderBottom: columnBorderStyle,
  },
  calendarColumn: {
    height: columnHeight,
    display: 'flex',
    flexDirection: 'column',
  },
});

/* istanbul ignore next */
function getDisplayDow(dow) {
  switch (dow) {
    case 'Mo':
      return 'Mon';
    case 'Tu':
      return 'Tue';
    case 'We':
      return 'Wed';
    case 'Th':
      return 'Thu';
    case 'Fr':
      return 'Fri';
    default:
      return '';
  }
}

export default function DowColumn({ dow }) {
  const classes = useStyles();
  return (
    <div className={classes.dowColumn}>
      <Typography className={classes.dowHeader} align="center" variant="body1">
        {getDisplayDow(dow)}
      </Typography>

      <div className={classes.calendarColumn}>
        {HOURS.map(
          hour => <HourCell key={hour} hour={hour} dow={dow} />,
        )}
      </div>
    </div>
  );
}

DowColumn.propTypes = {
  dow: PropTypes.string.isRequired,
};
