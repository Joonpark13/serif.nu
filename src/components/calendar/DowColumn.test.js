import React from 'react';
import { shallow } from 'enzyme';
import DowColumn from './DowColumn';

describe('DowColumn', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<DowColumn dow="Mon" />);

    expect(wrapper.get(0)).toMatchSnapshot();
  });
});
