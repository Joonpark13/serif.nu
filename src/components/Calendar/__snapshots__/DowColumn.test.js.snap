// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DowColumn renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DowColumn
    classes={
      Object {
        "calendarColumn": "",
        "dowColumn": "",
        "dowHeader": "",
      }
    }
    dow="Mon"
    hour={10}
    sections={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <WithStyles(Typography)
          align="center"
          className=""
          variant="body1"
        >
          Mon
        </WithStyles(Typography)>,
        <div
          className=""
        >
          <ForwardRef
            hour={8}
            sections={Array []}
          />
          <ForwardRef
            hour={9}
            sections={Array []}
          />
          <ForwardRef
            hour={10}
            sections={Array []}
          />
          <ForwardRef
            hour={11}
            sections={Array []}
          />
          <ForwardRef
            hour={12}
            sections={Array []}
          />
          <ForwardRef
            hour={13}
            sections={Array []}
          />
          <ForwardRef
            hour={14}
            sections={Array []}
          />
          <ForwardRef
            hour={15}
            sections={Array []}
          />
          <ForwardRef
            hour={16}
            sections={Array []}
          />
          <ForwardRef
            hour={17}
            sections={Array []}
          />
          <ForwardRef
            hour={18}
            sections={Array []}
          />
          <ForwardRef
            hour={19}
            sections={Array []}
          />
          <ForwardRef
            hour={20}
            sections={Array []}
          />
          <ForwardRef
            hour={21}
            sections={Array []}
          />
          <ForwardRef
            hour={22}
            sections={Array []}
          />
        </div>,
      ],
      "className": "",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "align": "center",
          "children": "Mon",
          "className": "",
          "variant": "body1",
        },
        "ref": null,
        "rendered": "Mon",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <ForwardRef
              hour={8}
              sections={Array []}
            />,
            <ForwardRef
              hour={9}
              sections={Array []}
            />,
            <ForwardRef
              hour={10}
              sections={Array []}
            />,
            <ForwardRef
              hour={11}
              sections={Array []}
            />,
            <ForwardRef
              hour={12}
              sections={Array []}
            />,
            <ForwardRef
              hour={13}
              sections={Array []}
            />,
            <ForwardRef
              hour={14}
              sections={Array []}
            />,
            <ForwardRef
              hour={15}
              sections={Array []}
            />,
            <ForwardRef
              hour={16}
              sections={Array []}
            />,
            <ForwardRef
              hour={17}
              sections={Array []}
            />,
            <ForwardRef
              hour={18}
              sections={Array []}
            />,
            <ForwardRef
              hour={19}
              sections={Array []}
            />,
            <ForwardRef
              hour={20}
              sections={Array []}
            />,
            <ForwardRef
              hour={21}
              sections={Array []}
            />,
            <ForwardRef
              hour={22}
              sections={Array []}
            />,
          ],
          "className": "",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "8",
            "nodeType": "function",
            "props": Object {
              "hour": 8,
              "sections": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "Naked": [Function],
              "displayName": "WithStyles(HourCell)",
              "options": Object {},
              "propTypes": Object {
                "classes": [Function],
                "innerRef": [Function],
              },
              "render": [Function],
              Symbol(Symbol.hasInstance): [Function],
            },
          },
          Object {
            "instance": null,
            "key": "9",
            "nodeType": "function",
            "props": Object {
              "hour": 9,
              "sections": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "Naked": [Function],
              "displayName": "WithStyles(HourCell)",
              "options": Object {},
              "propTypes": Object {
                "classes": [Function],
                "innerRef": [Function],
              },
              "render": [Function],
              Symbol(Symbol.hasInstance): [Function],
            },
          },
          Object {
            "instance": null,
            "key": "10",
            "nodeType": "function",
            "props": Object {
              "hour": 10,
              "sections": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "Naked": [Function],
              "displayName": "WithStyles(HourCell)",
              "options": Object {},
              "propTypes": Object {
                "classes": [Function],
                "innerRef": [Function],
              },
              "render": [Function],
              Symbol(Symbol.hasInstance): [Function],
            },
          },
          Object {
            "instance": null,
            "key": "11",
            "nodeType": "function",
            "props": Object {
              "hour": 11,
              "sections": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "Naked": [Function],
              "displayName": "WithStyles(HourCell)",
              "options": Object {},
              "propTypes": Object {
                "classes": [Function],
                "innerRef": [Function],
              },
              "render": [Function],
              Symbol(Symbol.hasInstance): [Function],
            },
          },
          Object {
            "instance": null,
            "key": "12",
            "nodeType": "function",
            "props": Object {
              "hour": 12,
              "sections": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "Naked": [Function],
              "displayName": "WithStyles(HourCell)",
              "options": Object {},
              "propTypes": Object {
                "classes": [Function],
                "innerRef": [Function],
              },
              "render": [Function],
              Symbol(Symbol.hasInstance): [Function],
            },
          },
          Object {
            "instance": null,
            "key": "13",
            "nodeType": "function",
            "props": Object {
              "hour": 13,
              "sections": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "Naked": [Function],
              "displayName": "WithStyles(HourCell)",
              "options": Object {},
              "propTypes": Object {
                "classes": [Function],
                "innerRef": [Function],
              },
              "render": [Function],
              Symbol(Symbol.hasInstance): [Function],
            },
          },
          Object {
            "instance": null,
            "key": "14",
            "nodeType": "function",
            "props": Object {
              "hour": 14,
              "sections": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "Naked": [Function],
              "displayName": "WithStyles(HourCell)",
              "options": Object {},
              "propTypes": Object {
                "classes": [Function],
                "innerRef": [Function],
              },
              "render": [Function],
              Symbol(Symbol.hasInstance): [Function],
            },
          },
          Object {
            "instance": null,
            "key": "15",
            "nodeType": "function",
            "props": Object {
              "hour": 15,
              "sections": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "Naked": [Function],
              "displayName": "WithStyles(HourCell)",
              "options": Object {},
              "propTypes": Object {
                "classes": [Function],
                "innerRef": [Function],
              },
              "render": [Function],
              Symbol(Symbol.hasInstance): [Function],
            },
          },
          Object {
            "instance": null,
            "key": "16",
            "nodeType": "function",
            "props": Object {
              "hour": 16,
              "sections": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "Naked": [Function],
              "displayName": "WithStyles(HourCell)",
              "options": Object {},
              "propTypes": Object {
                "classes": [Function],
                "innerRef": [Function],
              },
              "render": [Function],
              Symbol(Symbol.hasInstance): [Function],
            },
          },
          Object {
            "instance": null,
            "key": "17",
            "nodeType": "function",
            "props": Object {
              "hour": 17,
              "sections": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "Naked": [Function],
              "displayName": "WithStyles(HourCell)",
              "options": Object {},
              "propTypes": Object {
                "classes": [Function],
                "innerRef": [Function],
              },
              "render": [Function],
              Symbol(Symbol.hasInstance): [Function],
            },
          },
          Object {
            "instance": null,
            "key": "18",
            "nodeType": "function",
            "props": Object {
              "hour": 18,
              "sections": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "Naked": [Function],
              "displayName": "WithStyles(HourCell)",
              "options": Object {},
              "propTypes": Object {
                "classes": [Function],
                "innerRef": [Function],
              },
              "render": [Function],
              Symbol(Symbol.hasInstance): [Function],
            },
          },
          Object {
            "instance": null,
            "key": "19",
            "nodeType": "function",
            "props": Object {
              "hour": 19,
              "sections": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "Naked": [Function],
              "displayName": "WithStyles(HourCell)",
              "options": Object {},
              "propTypes": Object {
                "classes": [Function],
                "innerRef": [Function],
              },
              "render": [Function],
              Symbol(Symbol.hasInstance): [Function],
            },
          },
          Object {
            "instance": null,
            "key": "20",
            "nodeType": "function",
            "props": Object {
              "hour": 20,
              "sections": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "Naked": [Function],
              "displayName": "WithStyles(HourCell)",
              "options": Object {},
              "propTypes": Object {
                "classes": [Function],
                "innerRef": [Function],
              },
              "render": [Function],
              Symbol(Symbol.hasInstance): [Function],
            },
          },
          Object {
            "instance": null,
            "key": "21",
            "nodeType": "function",
            "props": Object {
              "hour": 21,
              "sections": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "Naked": [Function],
              "displayName": "WithStyles(HourCell)",
              "options": Object {},
              "propTypes": Object {
                "classes": [Function],
                "innerRef": [Function],
              },
              "render": [Function],
              Symbol(Symbol.hasInstance): [Function],
            },
          },
          Object {
            "instance": null,
            "key": "22",
            "nodeType": "function",
            "props": Object {
              "hour": 22,
              "sections": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "Naked": [Function],
              "displayName": "WithStyles(HourCell)",
              "options": Object {},
              "propTypes": Object {
                "classes": [Function],
                "innerRef": [Function],
              },
              "render": [Function],
              Symbol(Symbol.hasInstance): [Function],
            },
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <WithStyles(Typography)
            align="center"
            className=""
            variant="body1"
          >
            Mon
          </WithStyles(Typography)>,
          <div
            className=""
          >
            <ForwardRef
              hour={8}
              sections={Array []}
            />
            <ForwardRef
              hour={9}
              sections={Array []}
            />
            <ForwardRef
              hour={10}
              sections={Array []}
            />
            <ForwardRef
              hour={11}
              sections={Array []}
            />
            <ForwardRef
              hour={12}
              sections={Array []}
            />
            <ForwardRef
              hour={13}
              sections={Array []}
            />
            <ForwardRef
              hour={14}
              sections={Array []}
            />
            <ForwardRef
              hour={15}
              sections={Array []}
            />
            <ForwardRef
              hour={16}
              sections={Array []}
            />
            <ForwardRef
              hour={17}
              sections={Array []}
            />
            <ForwardRef
              hour={18}
              sections={Array []}
            />
            <ForwardRef
              hour={19}
              sections={Array []}
            />
            <ForwardRef
              hour={20}
              sections={Array []}
            />
            <ForwardRef
              hour={21}
              sections={Array []}
            />
            <ForwardRef
              hour={22}
              sections={Array []}
            />
          </div>,
        ],
        "className": "",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "align": "center",
            "children": "Mon",
            "className": "",
            "variant": "body1",
          },
          "ref": null,
          "rendered": "Mon",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <ForwardRef
                hour={8}
                sections={Array []}
              />,
              <ForwardRef
                hour={9}
                sections={Array []}
              />,
              <ForwardRef
                hour={10}
                sections={Array []}
              />,
              <ForwardRef
                hour={11}
                sections={Array []}
              />,
              <ForwardRef
                hour={12}
                sections={Array []}
              />,
              <ForwardRef
                hour={13}
                sections={Array []}
              />,
              <ForwardRef
                hour={14}
                sections={Array []}
              />,
              <ForwardRef
                hour={15}
                sections={Array []}
              />,
              <ForwardRef
                hour={16}
                sections={Array []}
              />,
              <ForwardRef
                hour={17}
                sections={Array []}
              />,
              <ForwardRef
                hour={18}
                sections={Array []}
              />,
              <ForwardRef
                hour={19}
                sections={Array []}
              />,
              <ForwardRef
                hour={20}
                sections={Array []}
              />,
              <ForwardRef
                hour={21}
                sections={Array []}
              />,
              <ForwardRef
                hour={22}
                sections={Array []}
              />,
            ],
            "className": "",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "8",
              "nodeType": "function",
              "props": Object {
                "hour": 8,
                "sections": Array [],
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "Naked": [Function],
                "displayName": "WithStyles(HourCell)",
                "options": Object {},
                "propTypes": Object {
                  "classes": [Function],
                  "innerRef": [Function],
                },
                "render": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
            },
            Object {
              "instance": null,
              "key": "9",
              "nodeType": "function",
              "props": Object {
                "hour": 9,
                "sections": Array [],
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "Naked": [Function],
                "displayName": "WithStyles(HourCell)",
                "options": Object {},
                "propTypes": Object {
                  "classes": [Function],
                  "innerRef": [Function],
                },
                "render": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
            },
            Object {
              "instance": null,
              "key": "10",
              "nodeType": "function",
              "props": Object {
                "hour": 10,
                "sections": Array [],
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "Naked": [Function],
                "displayName": "WithStyles(HourCell)",
                "options": Object {},
                "propTypes": Object {
                  "classes": [Function],
                  "innerRef": [Function],
                },
                "render": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
            },
            Object {
              "instance": null,
              "key": "11",
              "nodeType": "function",
              "props": Object {
                "hour": 11,
                "sections": Array [],
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "Naked": [Function],
                "displayName": "WithStyles(HourCell)",
                "options": Object {},
                "propTypes": Object {
                  "classes": [Function],
                  "innerRef": [Function],
                },
                "render": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
            },
            Object {
              "instance": null,
              "key": "12",
              "nodeType": "function",
              "props": Object {
                "hour": 12,
                "sections": Array [],
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "Naked": [Function],
                "displayName": "WithStyles(HourCell)",
                "options": Object {},
                "propTypes": Object {
                  "classes": [Function],
                  "innerRef": [Function],
                },
                "render": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
            },
            Object {
              "instance": null,
              "key": "13",
              "nodeType": "function",
              "props": Object {
                "hour": 13,
                "sections": Array [],
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "Naked": [Function],
                "displayName": "WithStyles(HourCell)",
                "options": Object {},
                "propTypes": Object {
                  "classes": [Function],
                  "innerRef": [Function],
                },
                "render": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
            },
            Object {
              "instance": null,
              "key": "14",
              "nodeType": "function",
              "props": Object {
                "hour": 14,
                "sections": Array [],
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "Naked": [Function],
                "displayName": "WithStyles(HourCell)",
                "options": Object {},
                "propTypes": Object {
                  "classes": [Function],
                  "innerRef": [Function],
                },
                "render": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
            },
            Object {
              "instance": null,
              "key": "15",
              "nodeType": "function",
              "props": Object {
                "hour": 15,
                "sections": Array [],
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "Naked": [Function],
                "displayName": "WithStyles(HourCell)",
                "options": Object {},
                "propTypes": Object {
                  "classes": [Function],
                  "innerRef": [Function],
                },
                "render": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
            },
            Object {
              "instance": null,
              "key": "16",
              "nodeType": "function",
              "props": Object {
                "hour": 16,
                "sections": Array [],
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "Naked": [Function],
                "displayName": "WithStyles(HourCell)",
                "options": Object {},
                "propTypes": Object {
                  "classes": [Function],
                  "innerRef": [Function],
                },
                "render": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
            },
            Object {
              "instance": null,
              "key": "17",
              "nodeType": "function",
              "props": Object {
                "hour": 17,
                "sections": Array [],
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "Naked": [Function],
                "displayName": "WithStyles(HourCell)",
                "options": Object {},
                "propTypes": Object {
                  "classes": [Function],
                  "innerRef": [Function],
                },
                "render": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
            },
            Object {
              "instance": null,
              "key": "18",
              "nodeType": "function",
              "props": Object {
                "hour": 18,
                "sections": Array [],
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "Naked": [Function],
                "displayName": "WithStyles(HourCell)",
                "options": Object {},
                "propTypes": Object {
                  "classes": [Function],
                  "innerRef": [Function],
                },
                "render": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
            },
            Object {
              "instance": null,
              "key": "19",
              "nodeType": "function",
              "props": Object {
                "hour": 19,
                "sections": Array [],
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "Naked": [Function],
                "displayName": "WithStyles(HourCell)",
                "options": Object {},
                "propTypes": Object {
                  "classes": [Function],
                  "innerRef": [Function],
                },
                "render": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
            },
            Object {
              "instance": null,
              "key": "20",
              "nodeType": "function",
              "props": Object {
                "hour": 20,
                "sections": Array [],
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "Naked": [Function],
                "displayName": "WithStyles(HourCell)",
                "options": Object {},
                "propTypes": Object {
                  "classes": [Function],
                  "innerRef": [Function],
                },
                "render": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
            },
            Object {
              "instance": null,
              "key": "21",
              "nodeType": "function",
              "props": Object {
                "hour": 21,
                "sections": Array [],
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "Naked": [Function],
                "displayName": "WithStyles(HourCell)",
                "options": Object {},
                "propTypes": Object {
                  "classes": [Function],
                  "innerRef": [Function],
                },
                "render": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
            },
            Object {
              "instance": null,
              "key": "22",
              "nodeType": "function",
              "props": Object {
                "hour": 22,
                "sections": Array [],
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "Naked": [Function],
                "displayName": "WithStyles(HourCell)",
                "options": Object {},
                "propTypes": Object {
                  "classes": [Function],
                  "innerRef": [Function],
                },
                "render": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
